src/osd/OSD.cc

```diff
@@ -6954,6 +6954,8 @@ void OSD::_committed_osd_maps(epoch_t first, epoch_t last, MOSDMap *m)
 
 	hbclient_messenger->mark_down_all();
 
+	heartbeat_set_peers_need_update();
+	last_heartbeat_resample = ceph_clock_now(cct);
```

Comment:
start_waiting_for_healthy() above sets last_heartbeat_resample to utime_t().  reset_heartbeat_peers() in that case already does this.